src/typeahead/typeahead.js

```diff
@@ -170,6 +171,52 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
         });
       };
 
+      // bind events only if appendToBody params exist - performance feature
+      if ( appendToBody ) {
+        angular.element( $window ).bind( 'resize', fireRecalculating );
+        $document.find( 'body' ).bind( 'scroll', fireRecalculating );
+      }
+
+      // Declare the timeout promise var outside the function scope so that stacked calls can be cancelled later
+      var timeoutEventPromise;
+
+      // Default progress type
+      scope.moveInProgress = false;
+
+      function fireRecalculating() {
+
+        if( !scope.moveInProgress ){
+          scope.moveInProgress = true;
+          scope.$digest();
+        }
+
+        // Cancel previous timeout
+        if ( timeoutEventPromise ) {
+          $timeout.cancel(timeoutEventPromise);
+        }
+
+        // Debounced executing recalculate after events fired
+        timeoutEventPromise = $timeout(function () {
+
+          // if popup is visible
+          if ( scope.matches.length ) {
+            recalculatePosition();
+          }
+
+          scope.moveInProgress = false;
+          scope.$digest();
+
+        }, eventDebounceTime);
+
+      }
+
+      // recalculate actual position and set new values to scope
+      // after digest loop is popup in right position
+      function recalculatePosition() {
+        scope.position = appendToBody ? $position.offset(element) : $position.position(element);
+        scope.position.top = scope.position.top + element.prop('offsetHeight');
```

Comment:
This is function copied from original code and top, must be calculate with offsetHeight of input, because dropdown must be showed after bottom edge of input.